File: finanger/lib/models/transactions.dart
class Transactions {
  final String tId;
  final String title;
  final String type;
  final String description;
  final String category;
  final double amount;
  final DateTime date;
  final String userId;

  Transactions({
    required this.tId,
    required this.title,
    required this.type,
    required this.description,
    required this.category,
    required this.amount,
    required this.date,
    required this.userId,
  });
}

File: backend/models/transactions.js
"use strict";
const { Model } = require("sequelize");
module.exports = (sequelize, DataTypes) => {
  class Transactions extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // define association here
      this.belongsTo(models.User, { foreignKey: "userId" });
      this.belongsTo(models.Categories, { foreignKey: "userId" });
    }
  }
  Transactions.init(
    {
      tId: DataTypes.STRING,
      title: DataTypes.STRING,
      type: DataTypes.STRING,
      description: DataTypes.STRING,
      category: DataTypes.STRING,
      amount: DataTypes.DOUBLE,
      date: DataTypes.DATE,
      userId: DataTypes.STRING,
    },
    {
      sequelize,
      modelName: "Transactions",
    }
  );
  return Transactions;
};
